ui/src/app/pages/data/concept/concept-table.tsx

```diff
@@ -263,15 +263,18 @@ export class ConceptTable extends React.Component<Props, State> {
   }
 
   selectAllHeader() {
-    if (this.state.showBanner) {
-      const {concepts} = this.props;
-      const bannerText = this.state.selectAll ? 'Clear Selection' : 'Select ' + concepts.length;
-      return <FlexRow  data-test-id='selection' style={{fontWeight: '200'}}>
-        All concepts on this page are selected.&nbsp;
+    const {concepts} = this.props;
+    if (this.state.showBanner && concepts.length > ROWS_TO_DISPLAY) {
+      const selectedConceptSize = this.state.selectAll ? concepts.length : ROWS_TO_DISPLAY;
```

Comment:
I'm confused by this. At this point in the code is there some kind of guarantee that the user has selected everything being displayed on the page? If `selectAll` is false, why do we assume they have selected `ROWS_TO_DISPLAY` concepts.

Reply:
the method updateSelectedConceptList(selectedConcepts, origin) sets the state variable showBanner to be true only if the user selects the checkbox in the header

Reply:
I think `showBanner` should be renamed to something like `allVisibleConceptsSelected`. Also, I can't reconcile the logic at L204 with what you're saying here.

Say I do the following - doesn't this work incorrectly?
1. Click select all checkbox
-> banner says "selected 20 concepts"
2. deselect one concept
 -> banner still says "selected 20 concepts"?